### ISSUE TITLE
Multiple usb configuration randomically resets in 6.3.3

### ISSUE THREAD

**Issue by TGD-Simracing** [zegreatclan/AssettoCorsaTools#229]
I often lose the multiple usb configuration with no apparent reason.
The check is reset and I have to activate it manually when this happens.

**Comment by zegreatclan** [comment]
I checked the code, absolutely nowhere except ui the check is changed. I feel you felt into the "developer" trap, if somehow you disabled it and got it saved, it will only be saved when simhub is properly closed (stop in visual studio is not a proper close :D).

If you close simhub properly however it would be interesting to get the log files. Also disabling plugins could be interesting, if somehow one plugin fails to stop it will prevent proper settings save.

Any other infos on how to reproduce it would be welcome.

**Comment by zegreatclan** [comment]
Any additionnal infos about my suggestions ?

**Comment by zegreatclan** [comment]
Any additionnal infos about my suggestions ?

**Comment by TGD-Simracing** [comment]
The only infos I have can be related to the custom protocol and the disconnection

of arduinos. Maybe if I disconnect a device while the multiple usb option is active and then I reconnect it, the multiple usb mode is not re-enabled.
I was wondering why the multiple-usb configuration is required to make the custom protocol works.
Thanks

Carlo


________________________________
Da: Wotever <notifications@github.com>
Inviato: martedì 27 febbraio 2018 16:37
A: zegreatclan/AssettoCorsaTools
Cc: Carlo Iovino; Author
Oggetto: Re: [zegreatclan/AssettoCorsaTools] Multiple usb configuration randomically resets in 6.3.3 (#229)


Any additionnal infos about my suggestions ?

**Comment by zegreatclan** [comment]
Multiple USB like I said is never disabled by simhub, only by user, but the enabled state is only saved at exit, so if somehow you enable it, and for a reason or another it exits badly (a plugin breaking the exit process for instance) it won't be saved and will be back as disabled at next start, but only log files can tell it, exit fails would be logged.

Concerning custom protocol the answer is in the doc ;) https://github.com/zegreatclan/AssettoCorsaTools/wiki/Custom-Arduino-hardware-support

Basically custom protocol is linked to a specific arduino, and only multiple usb offer a way to be certain to talk to the right arduino, today you use only a speedo , it may look overkill, but it could be a speedo an one arduino,a fuel gauge on another etc ....
